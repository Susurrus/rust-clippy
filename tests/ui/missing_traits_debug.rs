#![feature(untagged_unions)]
#![allow(dead_code)]
#![allow(clippy::all)]
#![warn(clippy::missing_debug_implementations)]

pub enum A {}

#[derive(Debug)]
pub enum B {}

pub enum C {}

impl std::fmt::Debug for C {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        f.debug_struct("C").finish()
    }
}

pub struct Foo;

#[derive(Debug)]
pub struct Bar;

pub struct Baz;

impl std::fmt::Debug for Baz {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        f.debug_struct("Baz").finish()
    }
}

struct PrivateStruct;

enum PrivateEnum {}

struct GenericType<T>(T);

fn main() {}
