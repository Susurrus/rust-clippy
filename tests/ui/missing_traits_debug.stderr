error: type does not implement `Debug`; consider adding #[derive(Debug)] or a manual implementation
  --> $DIR/missing_traits_debug.rs:6:1
   |
LL | pub enum A {}
   | ^^^^^^^^^^^^^
   |
   = note: `-D clippy::missing-debug-implementations` implied by `-D warnings`

error: type does not implement `Debug`; consider adding #[derive(Debug)] or a manual implementation
  --> $DIR/missing_traits_debug.rs:19:1
   |
LL | pub struct Foo;
   | ^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

